### PR TITLE
feat: KPI-234 change function to get metric values

### PR DESCRIPTION
**Pull request by FreddyJR1995**
## Jira Link

[Go to Jira](https://kpinetwork.atlassian.net/browse/KPI-234)

## Types of changes

- [ ] Hotfix (Bug detected)
- [x] Feature (Issue from an Active Sprint)

## Checklist:

- [x] Unit Test.
- [x] Coverage.

### CODE REVIEW

File: src/service/investment_date_report/investment_date_report.py [hunk 1]
```diff
@@ -70,43 +71,65 @@ def get_by_metric_records(
...
     def sort_metric_values(self, metric_values: dict) -> dict:
         metric = {int(k): v for k, v in metric_values.items()}
         return OrderedDict(sorted(metric.items()))
 
-    def get_companies_metrics(self, metrics: list, access: bool, **conditions) -> dict:
+    def get_companies(self, metrics: list, access: bool, **conditions) -> dict:
         companies_result = dict()
         investments = self.repository.get_investments()
         companies = investments.keys()
+        initial_year = 0
```
Comment (kellycastrof) [inline src/service/investment_date_report/investment_date_report.py:122]: This isn't needed, its value is always 0, please remove it.

File: src/service/investment_date_report/investment_date_report.py [hunk 2]
```diff
@@ -70,43 +71,65 @@ def get_by_metric_records(
...
         for metric in metrics:
             data = self.get_by_metric_records(
-                metric, years, access, company_id, **conditions
+                metric, years, access, companies_by_investment, **conditions
             )
-            company.update(data[0])
-            metrics_values.append(data[1])
-        company["metrics"] = metrics_values
-        return company
+            company_data = dict()
```
Comment (kellycastrof) [inline src/service/investment_date_report/investment_date_report.py:102]: Should this company_data be defined in this line?

File: src/service/investment_date_report/investment_date_report.py [hunk 3]
```diff
@@ -70,43 +71,65 @@ def get_by_metric_records(
...
+            for id in data:
+                metric_values = list()
+                company_data = data[id]
+                if id in companies:
+                    metric_values = companies[id]["metrics"]
+                metric_values.append(data[id]["metrics"])
+                companies[id] = company_data
```
Comment (kellycastrof) [inline src/service/investment_date_report/investment_date_report.py:109]: This company_data seems to be unnecessary, you can assign it directly with data[id]

File: src/service/investment_date_report/investment_date_report.py [hunk 4]
```diff
@@ -40,6 +40,7 @@ def __init__(
 
     def get_valid_year_records(self, company: dict, years: list) -> dict:
         metric_years = [*company["metrics"]]
+        included_years = set(metric_years) & set(years)
```
Comment (kellycastrof) [inline src/service/investment_date_report/investment_date_report.py:43]: Why is used this line? Please remove it

**Review by kellycastrof — CHANGES_REQUESTED** [review]

File: src/service/investment_date_report/investment_date_report.py (hunk 2 above)
Comment (FreddyJR1995) [reply src/service/investment_date_report/investment_date_report.py]: It was deleted.

File: src/service/investment_date_report/investment_date_report.py (hunk 3 above)
Comment (FreddyJR1995) [reply src/service/investment_date_report/investment_date_report.py]: With `company_data `deleted the value was changed to `data[id]`.

**Review by kellycastrof — APPROVED** [review]

**Pull request merged**